### PR TITLE
Fixed resultSet typing for rows

### LINKED ISSUE #1996: Fix incorrect typing in `ResultsetTable`
### Discussed in https://github.com/dxc-technology/halstack-react/discussions/1981

<div type='discussions-op-text'>

<sup>Originally posted by **Mil4n0r** May  8, 2024</sup>
The existing `Row` type is counterintuitive with its meaning.

### PR DESCRIPTION
**Pull request by Mil4n0r**
**Checklist**
_(Check off all the items before submitting)_

- [x] Build process is done without errors. All tests pass in the `/lib` directory.
- [ ] Self-reviewed the code before submitting.
- [x] Meets accessibility standards.
- [x] Added/updated documentation to `/website` as needed.
- [ ] Added/updated tests as needed.

**Description**
Typing for rows inside resultsetTable was incorrect.

**Closes #1996**

### CODE REVIEW

File: lib/src/resultset-table/types.ts [hunk 1]
```diff
@@ -29,6 +29,8 @@ export type Row = {
   sortValue?: string;
 };
 
+export type Row = Cell[];
```
Comment (Jialecl) [inline lib/src/resultset-table/types.ts:32]: I think there is no need to export the type `Row`.

File: website/screens/components/resultset-table/code/ResultsetTableCodePage.tsx [hunk 2]
```diff
@@ -81,12 +81,12 @@ const sections = [
             </td>
             <td>
               <TableCode>
-                {"{ displayValue: React.ReactNode; sortValue?: string; }[]"}
+                {"{ displayValue: React.ReactNode; sortValue?: string; }[][]"}
```
Comment (Jialecl) [inline website/screens/components/resultset-table/code/ResultsetTableCodePage.tsx:84]: While the typing is correct it is not very visual, maybe we can make better use of the description to improve this.
We can type it as Row[] and explain int the description:

`{ displayValue: React.ReactNode; sortValue?: string; } = Cell`
`Cell[] = Row`

Comment (Mil4n0r) [reply website/screens/components/resultset-table/code/ResultsetTableCodePage.tsx]: Sounds like a good idea, can you take a look at it now?

File: website/screens/components/resultset-table/code/ResultsetTableCodePage.tsx [hunk 3]
```diff
@@ -31,6 +31,9 @@ const actionsType = `{
   options: Option[];
 }[]`;
 
+const cellTypeString = `{ displayValue: React.ReactNode; sortValue?: string; }[]`;
```
Comment (Jialecl) [inline website/screens/components/resultset-table/code/ResultsetTableCodePage.tsx:34]: This is too long. 
Add line breaks.

**Review by Jialecl — APPROVED** [review]

**Pull request merged**